### ISSUE TITLE
[TIP001] Proposal for Standardized Mixin, Transaction Size, & Transaction Fees for Large Sends

### ISSUE THREAD

**Issue by RocksteadyTC** [turtlecoin/meta#59]
### Abstract
Setting logical default parameters for mixin, max transaction size, and fees ensure that a transaction is never excluded from being processed within a few blocks.

### Context
Recently we've grown a bit as a network, and as larger amounts of money are being transacted, larger amounts are easier to mix sufficiently. We're no longer at the stage where a mixin of 0 makes sense, from a technical perspective. Mixin 0 has historically also been used to clear large transactions in the Transaction Pool Queue so that they aren't waiting to be mixed for long periods. 

### Problem
Mixin 0 affords no privacy, is often a large transaction prone to getting stuck, and can possibly help deanonymize ongoing and past transactions when combined with other weakly mixed transactions. Currently there are quite a few large transactions of around 4MM and higher sitting in the pool waiting to be cleared with little or no fee, with a very small chance of ever being processed. Wallets currently allow excessively large transactions to be broadcast, even when the network has little chance of accepting them as valid.

### Proposed Solution
The solution to this issue is multi-faceted, and addresses the many variables that determine when and why certain transactions are being excluded from blocks.

- **Minimum Mixin** 
I propose we enforce a minimum mixin of 5 on all transactions. This should be easy regardless of what the maximum send amount would be decided to be, so long as there is one for each transaction. The upper bounds on tested mixins for small transactions that I've used are over Mixin 200+.  Today, I can send a 2MM TRTL transaction via mixin 5 and a fee of 200 TRTL without effort. The reasons for using a 0 mixin transaction are close to zero.

- **Max Transaction Size** 
Currently, out of the current stack of transactions sitting in the Transaction Pool Queue, all have a size of 117000 or more, up to 124000. These can be small transactions with lots of small inputs, usually from mining payouts, or these can be large transactions with low fees and large inputs. The cap should not be based on how much TRTL, but on the resulting number of inputs, or the size of the inputs so that no transaction passes a limit of 100,000 until the next upgrade. This limit should move upward as that transactional volume of the network grows, until we begin to see stuck transactions begin to appear again. 

- **Proportional Fees** 
Sending small transactions is not a problem, and these should pay no extra fee. Anything under 100,000 can send with a logical mixin for a fee of 0.1 TRTL, or the network minimum for forming a transaction. Anything above this amount, approaching the transaction size cap should pay a fee at a rate of 0.01% per transaction, for example, a transaction of 2MM TRTL would pay 200 TRTL as fee to be sent. 

### Alternatives
An alternative would be a modification to simplewallet and walletd to break up transactions containing inputs exceeding the maximum transaction size. The same modification could automatically apply a fee, but assuming the transaction could be broken up into small enough sends, an additional fee incentive may not even be necessary. 

### Notes & Considerations
The code for minimum mixin, and max transaction size would be trivial to implement, as I've been experimenting with this already independently. However, I'm not sure how to go about implementing a scaling fee structure, or that I'd be the right one to implement that. The first two objectives listed could be implemented as early as the next release window on Feb 18th 2018.

**Comment by SoreGums** [comment]
A discussion point.

At the network level what are the rules for a transaction being valid and thus processed into a block? Sure A miner can just flatout deny the acceptance of a transaction and effectively ignore it, that doesn't stop another party going "heh, I'll include you buddy". I send low fee transactions on Bitcoin and one of the miners simply doesn't care and just includes everything regardless of the fee, take a while, I don't mind waiting 10hrs vs 1hr...

My point is this proposal seems to be coming up with sensible defaults on the wallet side, and wallets don't decide if a transaction is to be included in a block or not. So lets make sure some thought goes into the network side of the identified problem. Anyone can compile a wallet that ignores the sensible defaults and does their own thing, as long they are producing network acceptable transactions everything is fine for them, they'll adjust params to the point that their transactions are accepted and that is in the spirit of the project I would think.

Basically we don't want a griefer to come in and spam the network with junk transactions that hang out in the mempool forever - so lets make sure the network can deal with incompatible transactions as well 🍕

**Comment by RocksteadyTC** [comment]
An update on this thread, with transaction splitting helping out on the stuck TX and fee side, and with the increase in network volume keeping the block size large and the mem pool clear, it seems wise to knock out the last item, which is standard mixin. 
I think we should start with 3 as a minimum, see where that goes, see what kinds of transactions get stuck or have trouble, and then proceed to bring up the mix potentially every upgrade after. I am not sure where the 'sweet spot' is regarding mixin, in a perfect world where everyone uses the same setting, but I'd imagine somewhere around 20-40 would be great, with 13-17 being more logically attainable.

**Comment by RocksteadyTC** [comment]
i think 3 is a good idea

**Comment by RocksteadyTC** [comment]
This can be closed.
